src/odemis/acq/stream/_sync.py

```diff
@@ -2420,6 +2434,17 @@ def _runAcquisition(self, future):
                     raise ValueError("Failed to configure emitter resolution to %s, got %s" %
                                      ((n_x, n_y), em_res))
 
+                # Update the resolution of the "independent" detectors
+                for s in self._streams:
+                    det = s._detector
+                    if model.hasVA(det, "resolution"):
+                        det.resolution.value = (n_x, n_y)
+                        if det.resolution.value != (n_x, n_y):
```

Comment:
This reads quite funnily. I suppose the setter can prevent the actual setting of the value, but it is not really trivial from just reading the code. Maybe comment why you are anticipating a failure of setting the value.